library/core/src/slice/index.rs

```diff
@@ -449,3 +451,104 @@ unsafe impl<T> SliceIndex<[T]> for ops::RangeToInclusive<usize> {
         (0..=self.end).index_mut(slice)
     }
 }
+
+#[stable(feature = "slice_index_bound_pair", since = "1.51.0")]
+unsafe impl<T> SliceIndex<[T]> for (ops::Bound<usize>, ops::Bound<usize>) {
+    type Output = [T];
+
+    #[inline]
+    fn get(self, slice: &[T]) -> Option<&[T]> {
+        let start = match self.0 {
+            ops::Bound::Unbounded => 0,
+            ops::Bound::Included(start) => start,
+            ops::Bound::Excluded(start) => start.checked_add(1)?,
+        };
+        let end = match self.1 {
+            ops::Bound::Unbounded => slice.len(),
+            ops::Bound::Included(end) => end.checked_add(1)?,
+            ops::Bound::Excluded(end) => end,
+        };
+        (start..end).get(slice)
+    }
+
+    #[inline]
+    fn get_mut(self, slice: &mut [T]) -> Option<&mut [T]> {
+        let start = match self.0 {
+            ops::Bound::Unbounded => 0,
+            ops::Bound::Included(start) => start,
+            ops::Bound::Excluded(start) => start.checked_add(1)?,
+        };
+        let end = match self.1 {
+            ops::Bound::Unbounded => slice.len(),
+            ops::Bound::Included(end) => end.checked_add(1)?,
+            ops::Bound::Excluded(end) => end,
+        };
+        (start..end).get_mut(slice)
+    }
+
+    #[inline]
+    unsafe fn get_unchecked(self, slice: *const [T]) -> *const [T] {
+        let start = match self.0 {
+            ops::Bound::Unbounded => 0,
+            ops::Bound::Included(start) => start,
+            ops::Bound::Excluded(start) => start + 1,
```

Comment:
I am not sure if it is a good idea to implicitly overflow in `get_unchecked` (and `get_mut`unchecked`). The "unchecked" refers to the slice bounds, how is the caller supposed to know that they also need to guard against overflow?

Reply:
So I used the existing code in `RangeInclusive` for precedent as that also doesn't guard against overflow. I'm fine just doing a saturated_add instead though.

Reply:
Hm, that's a fair point. But I still feel like this choice (either way) should be documented somewhere.